ironfish/src/mining/poolDatabase/database.ts

```diff
@@ -133,6 +133,17 @@ export class PoolDatabase {
 
     return result.count
   }
+
+  async getCurrentPayoutPeriod(): Promise<DatabasePayoutPeriod | undefined> {
+    return await this.db.get<DatabasePayoutPeriod>(
+      'SELECT * FROM payoutPeriod WHERE end is null',
+    )
+  }
+
+  async rolloverPayoutPeriod(timestamp: number): Promise<void> {
+    await this.db.run('UPDATE payoutPeriod SET end = ? WHERE end IS NULL', timestamp - 1)
+    await this.db.run('INSERT INTO payoutPeriod (start) VALUES (?)', timestamp)
```

Comment:
Do we have a way to put these two statements in a transaction? I can't think of why the insertion would fail, but it looks like we'd end up with a gap in payoutPeriods if it did.